File: templates/python/basic_query.py
"""Basic Claude Agent SDK query example."""
import asyncio
from claude_agent_sdk import (
    query, ClaudeAgentOptions,
    AssistantMessage, ResultMessage, TextBlock,
)

async def main():
    options = ClaudeAgentOptions(
        system_prompt="You are a helpful assistant.",
        max_turns=5,
        permission_mode="bypassPermissions",
    )

    async for message in query(prompt="What is 2 + 2?", options=options):
        if isinstance(message, AssistantMessage):
            for block in message.content:
                if isinstance(block, TextBlock):
                    print(block.text)
        elif isinstance(message, ResultMessage):
            if message.subtype == "success":
                print(f"Result: {message.result}")
                print(f"Cost: ${message.total_cost_usd or 0:.4f}")
            else:
                print(f"Error: {message.subtype}")

asyncio.run(main())

File: templates/typescript/basic-query.ts
import { query } from "@anthropic-ai/claude-agent-sdk";

/**
 * Basic Query Template
 *
 * Demonstrates:
 * - Simple query execution
 * - Message type handling
 * - Session ID capture
 */

async function basicQuery() {
  let sessionId: string | undefined;

  for await (const message of query({
    prompt: "Analyze the codebase and suggest improvements",
    options: {
      allowedTools: ["Read", "Grep", "Glob"],
      permissionMode: "bypassPermissions",
      allowDangerouslySkipPermissions: true,
    },
  })) {
    switch (message.type) {
      case "system":
        if (message.subtype === "init") {
          sessionId = message.session_id;
          console.log(`Session: ${sessionId}`);
          console.log(`Model: ${message.model}`);
          console.log(`Tools: ${message.tools.join(", ")}`);
        }
        break;

      case "assistant":
        // message.message is the full Anthropic API message object
        console.log("Assistant response received");
        break;

      case "result":
        if (message.subtype === "success") {
          console.log(`Result: ${message.result}`);
          console.log(`Cost: $${message.total_cost_usd.toFixed(4)}`);
          console.log(`Turns: ${message.num_turns}`);
        } else {
          console.error(`Error: ${message.subtype}`, message.errors);
        }
        break;
    }
  }
}

basicQuery().catch(console.error);
